### PR TITLE
ci: Implement coverage on top of mkosi 

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

File: mkosi.conf.d/10-arch/mkosi.conf
Comment (bluca) [inline mkosi.conf.d/10-arch/mkosi.conf]: doesn't seem to be used?

**Comment by evverx** [comment]
(I understand it's a draft and it's probably waiting for @mrc0mmand review but I don't think it should collide with CentOS CI on Coveralls until the switch takes place on CentOS CI. It of course makes sense to send data from this PR because Coveralls is going to compare mkosi coverage reports with the main branch so it's going to be possible to compare mkosi with CentOS CI to see what's missing and tweak the test suite accordingly)

File: src/shared/creds-util.c
Comment (YHNdnzj) [inline src/shared/creds-util.c]: SD_ID128_FORMAT_STR + _VAL?

**Review by bluca — APPROVED** [review]

File: mkosi.conf.d/05-tools/mkosi.conf
Comment (yuwata) [inline mkosi.conf.d/05-tools/mkosi.conf]: Not sorted.

Comment (daandemeyer) [reply mkosi.conf.d/05-tools/mkosi.conf]: I'll fix this in a follow up PR to not have to wait 3 hours for CI again.

**Comment by mrc0mmand** [comment]
Thanks! The current TEST-38-FREEZER fail happens in CentOS CI from time to time as well (the whole system just... freezes), but I've never gotten around to actually debug it. Once this is merged I'll disable the CentOS CI job.

**Comment by DaanDeMeyer** [comment]
> Thanks! The current TEST-38-FREEZER fail happens in CentOS CI from time to time as well (the whole system just... freezes), but I've never gotten around to actually debug it. Once this is merged I'll disable the CentOS CI job.

I added a skip for now, I can reproduce this locally but since everything's frozen it's rather hard to reproduce.

**Comment by evverx** [comment]
Looks like the fuzz target aren't covered for some reason:

(0%) https://coveralls.io/builds/71149771/source?filename=src%2Fsrc%2Flibsystemd-network%2Ffuzz-dhcp-server.c
(94.23%) https://coveralls.io/builds/71134784/source?filename=src%2Flibsystemd-network%2Ffuzz-dhcp-server.c

The path seems to be mangled as well (with `/src/src` instead of `/src`) (and that's probably why Coveralls can't compare the reports automatically)

**Comment by evverx** [comment]
Also I'm 99% sure there should have been a Coveralls status check with some stats and a link to the report. Maybe
```sh
    permissions:
      contents: read
      checks: write
```
can work (depending on how Coveralls is integrated)

**Comment by evverx** [comment]
Looks like the coveralls status didn't show up. @mrc0mmand I wonder if you have unflipped "USE STATUS API?" at some point https://docs.coveralls.io/app-notifications#status-updates?

**Comment by mrc0mmand** [comment]
> Looks like the coveralls status didn't show up. @mrc0mmand I wonder if you have unflipped "USE STATUS API?" at some point https://docs.coveralls.io/app-notifications#status-updates?

I didn't, but when I tried to check it I noticed that the integration between Coveralls and systemd organization is broken. I resynced it and can again access the settings, so maybe it'll work this time?

**Comment by DaanDeMeyer** [comment]
I'll deal with the rest of the fuzzers in a separate PR; the entire setup for tests/fuzz is completely crazy if you ask me (nested invocation of meson is completely bonkers).

**Comment by evverx** [comment]
> the entire setup for tests/fuzz is completely crazy if you ask me (nested invocation of meson is completely bonkers).

As far as I can remember the idea was to make it possible to run the fuzz targets as part of the regression test suite under ASan/UBSan regardless of how systemd itself is compiled because those bugs aren't always reproducible with plain builds. I don't remember where it came from though. All my kludges went to `tools/oss-fuzz.sh` usually :-)

**Comment by DaanDeMeyer** [comment]
I removed the logic to run this on PRs, setting the green label.

**Comment by evverx** [comment]
For the record the diff between the last CentOS CI build and the first mkosi build cant be found at https://coveralls.io/builds/71222882

**Pull request merged**
